crates/biome_js_syntax/src/import_ext.rs

```diff
@@ -303,4 +303,37 @@ impl AnyJsImportSpecifierLike {
             }
         }
     }
+
+    /// Check whether the js import specifier like is in a ts module declaration:
+    /// ```ts
+    /// declare "abc" {}
+    /// ```
+    /// ## Examples
+    ///
+    /// ```
+    /// use biome_js_factory::make;
+    /// use biome_js_syntax::{AnyJsImportSpecifierLike, JsSyntaxKind, JsSyntaxToken};
+    /// use biome_rowan::TriviaPieceKind;
+    ///
+    /// let module_token = JsSyntaxToken::new_detached(JsSyntaxKind::MODULE_KW, "module", [], []);
+    /// let module_source = make::js_module_source(make::js_string_literal("foo"));
+    /// let module_declaration = make::ts_external_module_declaration(module_token, module_source).build();
+    /// let any_import_specifier = AnyJsImportSpecifierLike::JsModuleSource(module_declaration.source().expect("module source"));
+    /// assert!(any_import_specifier.is_in_ts_module_declaration());
+    ///
+    /// let module_source = make::js_module_source(make::js_string_literal("bar"));
+    /// let any_import_specifier = AnyJsImportSpecifierLike::JsModuleSource(module_source);
+    /// assert!(!any_import_specifier.is_in_ts_module_declaration());
+    /// ```
+    pub fn is_in_ts_module_declaration(&self) -> bool {
+        // It first has to be a JsModuleSource
```

Comment:
How about changing this method to `is_in_import`? Because for `AnyImportSpecifierLike`, that's the only thing we care about.

Reply:
I get your point but I think `is_in_ts_module_declaration` can reflect the real implementation of this function, and maybe we shouldn't anticipate too much of how this will be used in the future.

Reply:
That can also simplify the `module_name_token` and `inner_string_text` methods, as part of their logic involves checking if they are in the `import`

https://github.com/biomejs/biome/blob/549a8699e91070913b35d1534f95a5976986b47d/crates/biome_js_syntax/src/import_ext.rs#L228-L232

Reply:
I'm not sure I follow you. The rules this PR fixes also account for `require` calls, so I don't see how an `is_in_import` will help here. And ts module declaration also seems not an import to me. But there're definitely duplicate logics that can be abstracted away. Those can be addressed in furture PRs I think.